### ISSUE TITLE
Add CI jobs for each X86 instruction set architecture

### ISSUE THREAD

**Issue by fiigii** [dotnet/coreclr#21339]
We have exposed all the X86 ISA knobs (like `COMPlus_EnableFMA`) in release build, but CI just has a subset.
https://github.com/dotnet/coreclr/blob/a089f64a7ad55a8ce0b3203b1bf87040775dff01/netci.groovy#L98-L104

We should add CI jobs for each X86 ISA knobs for testing target-specific code like https://github.com/dotnet/coreclr/pull/21073

**Comment by fiigii** [comment]
cc @CarolEidt @tannergooding @BruceForstall

**Comment by tannergooding** [comment]
We can also simplify a couple of the checks (such as `jitnox86hwintrinsic`)  now that we have them strictly hierarchical from base up

**Comment by fiigii** [comment]
Yes, we can use the new knob `EnableHWIntrinsic` that you added recently.

**Comment by fiigii** [comment]
Good to know, thanks.

**Comment by tannergooding** [comment]
@BruceForstall, do we have an ETA on the conversion?

It would be nice to have this additional validation enabled sufficiently before the feature ships to help validate correctness.

**Comment by BruceForstall** [comment]
Maybe @echesakovMSFT @RussKeldorph can share a timeline (I don't know the ETA).

**Comment by echesakovMSFT** [comment]
@tannergooding I am planning to merge in https://github.com/dotnet/coreclr/pull/21966 later today or tomorrow, so this should unblock you from defining new test scenarios for Azure DevOps. 

Just add the corresponding TestEnvironment items to testenvironment.proj file.
